src/Models/Rate.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tipoff\Bookings\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Rate extends Model
+{
+    use HasFactory;
+
+    protected $guarded = ['id'];
+    protected $casts = [
+    ];
+
+    protected static function boot()
+    {
+        parent::boot();
+
+        static::creating(function ($rate) {
+            if (auth()->check()) {
+                $rate->creator_id = auth()->id();
+            }
+        });
+
+        static::saving(function ($rate) {
+            if (auth()->check()) {
+                $rate->updater_id = auth()->id();
+            }
+            if (empty($rate->slug)) {
+                throw new \Exception('A rate must have a slug.');
+            }
+            if (empty($rate->amount)) {
```

Comment:
Tried to keep as much as the original logic, but updated here to match the migration schema.

Reply:
Can you please change any foreign reference to Interface approaches? https://github.com/tipoff/support/pull/85/files#diff-2446fbd18a48df9f178dcebd7492e4ce75e9588ef0cafad2130f19a6b7b7fcf6R30

So `->getAmount()` instead. I personally like eloquent way to access data bot because its all distributed between so many packages we should focus on interfaces.

Reply:
https://github.com/tipoff/bookings/pull/47/commits/458f6009d7644199c7778ef97069936cb9f2dc17